src/deepsparse/pipelines/embedding_extraction.py

```diff
@@ -0,0 +1,224 @@
+
+from deepsparse import Pipeline
+from deepsparse.log import get_main_logger
+from deepsparse.utils import truncate_onnx_embedding_model
+
+
+__all__ = [
+    "EmbeddingExtractionOutput",
+    "EmbeddingExtractionPipeline",
+]
+
+
+_LOGGER = get_main_logger()
+
+
+class EmbeddingExtractionOutput(BaseModel):
+    """
+    Schema for embedding_extraction pipeline output. Values are in batch order
+    """
+
+    # List[Any] is for accepting numpy arrays
+    embeddings: Union[List[List[float]], List[Any]] = Field(
+        description="The output of the model which is an embedded "
+        "representation of the input"
+    )
+
+    class Config:
+        arbitrary_types_allowed = True
+
+
+@Pipeline.register(
+    task="embedding_extraction",
+    task_aliases=[],
+)
+class EmbeddingExtractionPipeline(Pipeline):
+    """
+    embedding extraction pipeline for extracting intermediate layer embeddings
+
+    :param model_path: sparsezoo stub to a transformers model or (preferred) a
+        directory containing a model.onnx, tokenizer config, and model config
+    :param engine_type: inference engine to use. Currently supported values include
+        'deepsparse' and 'onnxruntime'. Default is 'deepsparse'
+    :param batch_size: static batch size to use for inference. Default is 1
+    :param num_cores: number of CPU cores to allocate for inference engine. None
+        specifies all available cores. Default is None
+    :param scheduler: (deepsparse only) kind of scheduler to execute with.
+        Pass None for the default
+    :param input_shapes: list of shapes to set ONNX the inputs to. Pass None
+        to use model as-is. Default is None
+    :param alias: optional name to give this pipeline instance, useful when
+        inferencing with multiple models. Default is None
+    :param emb_extraction_layer: if an int, the layer number from which
+        the embeddings will be extracted. If a string, the name of last
+        ONNX node in model to draw embeddings from. If None, leave the model
+        unchanged. Default is None
+    :param flatten_outputs: if True, embeddings will be flattened along the batch (0)
+        dimension. Default False
+    :param return_numpy: return embeddings a list of numpy arrays, list of lists
+        of floats otherwise. Default is False
+    """
+
+    def __init__(
+        self,
+        *,
+        base_task: str,
+        emb_extraction_layer: Union[int, str, None] = None,
+        return_numpy: bool = False,  # to support Pydantic Validation
+        flatten_outputs: bool = False,
+        **base_pipeline_args,
+    ):
+        self._base_task = base_task
+        self._emb_extraction_layer = emb_extraction_layer
+        self._return_numpy = return_numpy
+        self._flatten_outputs = flatten_outputs
+
+        # initialize engine after model truncate
+        self.base_pipeline = Pipeline.create(
+            task=base_task,
+            _delay_engine_initialize=True,
```

Comment:
If the main purpose of this is to avoid doing the expensive model compilation, maybe we can explicitly set the engine_type to ORT here, and then revert it back to its default before calling _initialize_engine() below.

I think that should remove the need for this internal delay flag

Reply:
both approaches have small drawbacks, i'm in favor of leaving as is for now and revisiting later to clean up

Reply:
@corey-nm I've made a note of this and will come back with a better solution at a later time, for now leaving this as is unless this is a blocker from your side